src/components/utils/ComboBoxItems/ComboBoxItems.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { GetItemPropsOptions } from 'downshift';
+import Text from '../../Text';
+import Box from '../../Box';
+import MenuItem from '../MenuItem';
+import groupBy from 'lodash/groupBy';
+
+interface ComboBoxItemsProps<T> {
+  items: T[];
+  disableItem: (item: T) => boolean;
+  getItemProps: (options: GetItemPropsOptions<T>) => T;
+  itemToString: (item: T) => string;
+  itemToGroup?: (item: T) => string;
+  selectedItem?: T | null;
+}
+
+/**
+ * Acts as a wrapper of MenuItems in a list of dropdown options,
+ * to allow grouping.
+ */
+function ComboBoxItems<Item>({
+  items,
+  itemToGroup,
+  ...rest
+}: ComboBoxItemsProps<Item>): React.ReactElement<ComboBoxItemsProps<Item>> {
+  const groupedItems = React.useMemo(() => {
+    if (!itemToGroup) {
+      return [];
+    }
```

Comment:
optional: we can remove this line since it's never used. If there's no `itemToGroup` then you return early in the statement below and you don't us e`groupedItems` at all.